packages/shared/friends/sagas.ts

```diff
@@ -1389,6 +1390,46 @@ export async function getChannelMessages(request: GetChannelMessagesPayload) {
   getUnityInstance().AddChatMessages(addChatMessages)
 }
 
+// Search channels
+export async function searchChannels(request: GetChannelsPayload) {
+  const client: SocialAPI | null = getSocialClient(store.getState())
+  if (!client) return
+
+  const searchTerm: string | undefined = request.name === '' ? undefined : request.name
+  const since: string | undefined = request.since === '' ? undefined : request.since
+
+  // get user joined channelIds
+  const joinedChannelIds = getAllConversationsWithMessages(store.getState())
+    .filter((conv) => conv.conversation.type === ConversationType.CHANNEL)
+    .map((conv) => conv.conversation.id)
+
+  const profile = getCurrentUserProfile(store.getState())
+
+  // search channels
+  const { channels, nextBatch } = await client.searchChannel(request.limit, searchTerm, since)
+
+  const channelsToReturn = channels.map((channel) => ({
```

Comment:
does `channelsToReturn` have a type assertion to ensure we are not accidentally breaking the data type expected by unity?

Reply:
Added explicit types to `channelsToReturn` and `searchResult` if that's what you meant